lib/vpc/manager/Manager.rb

```diff
@@ -93,11 +93,162 @@ def update(local, diffs)
         end
       end
 
+      # Public: Renames an asset and all references to it and syncs the name tag change to AWS.
+      def rename(asset_type, old_name, new_name)
+
+        # Strip .json from the old_name and new_name
+        old_name = old_name.end_with?(".json") ? old_name[0...-5] : old_name
+        new_name = new_name.end_with?(".json") ? new_name[0...-5] : new_name
+
+        vpcs_dir = Configuration.instance.vpc.vpcs_directory
+        subnets_dir = Configuration.instance.vpc.subnets_directory
+        route_tables_dir = Configuration.instance.vpc.route_tables_directory
+        policies_dir = Configuration.instance.vpc.policies_directory
+        network_acls_dir = Configuration.instance.vpc.network_acls_directory
+
+        case asset_type
+        when "network-acl"
+
+          puts Colors.blue("Renaming Network ACL #{old_name} to #{new_name}")
+
+          # Update the Name tag and resave the file
+          acl_local = Loader.network_acl(old_name)
+          acl_local.tags["Name"] = new_name
+          json = JSON.pretty_generate(acl_local.to_hash)
+          File.open("#{network_acls_dir}/#{new_name}.json", "w") { |f| f.write(json) }
+
+          # Update the tags in AWS
+          acl_aws = EC2::named_network_acls[old_name]
+          create_tags(acl_aws.network_acl_id, { "Name" => new_name})
```

Comment:
The original file is only deleted at the end, and yeah they would probably just `reset HEAD --hard` or something. The only thing that can fail because of AWS is the tag creation, which happens right away... the rest of it should be pretty harmless since it's all local file changes